### ISSUE TITLE
Unable to build gradleTest project with Gradle 2.1+

### ISSUE THREAD

**Issue by szpak** [ysb33r/gradleTest#26]
When Gradle wrapper bumped to version 2.1 (tested with 2.1, 2.6 and 2.12) a try to build the gradleTest project itself fails with the followed error:
```
$ gw clean check
Download https://services.gradle.org/distributions/gradle-2.1-bin.zip
[buildinfo] Not using buildInfo properties file for this build.
:clean
:compileJava UP-TO-DATE
:compileGroovy
startup failed:
/home/szpak/Code/inne/gradleTest/src/main/groovy/org/ysb33r/gradle/gradletest/internal/DistributionInternal.groovy: 33: Can't have an abstract method in a non-abstract class. The class 'org.ysb33r.gradle.gradletest.internal.DistributionInternal' must be declared abstract or the method 'int compareTo(java.lang.Object)' must be implemented.
 @ line 33, column 1.
   @TupleConstructor
   ^

1 error

:compileGroovy FAILED
```

Build with OpenJDK 1.8.0_72.

**Comment by ysb33r** [comment]
This is intended to be built with `2.0`, but tested against as many following gradle versions, as to ensure widest compatibility. I have not looked into building it with any further version - will probably not be done until I start the integration with `TestKit`.

**Comment by szpak** [comment]
The problem with Gradle 2.0 is that IntelliJ Idea cannot import projects configured to use it when OpenJDK is used (was fixed in Gradle 2.1 which fails in your project and it is tricky to import and use your project in Idea with OpenJDK). Nevertheless I can wait until #33 is implemented.

**Comment by ysb33r** [comment]
OK, if I get a chance I'll look into it.
